File: bindings/csharp/src/ProvenTls.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// TLS protocol bindings for proven-servers.

namespace Proven;

/// <summary>TlsState matching the Idris2 ABI tags (0-6).</summary>
public enum TlsState : byte
{
    TlsIdle = 0,
    TlsClientHello = 1,
    TlsServerHello = 2,
    TlsNegotiating = 3,
    TlsEstablished = 4,
    TlsRenegotiating = 5,
    TlsShutdown = 6
}

/// <summary>TlsVersion matching the Idris2 ABI tags (0-1).</summary>
public enum TlsVersion : byte
{
    Tls12 = 0,
    Tls13 = 1
}

/// <summary>CipherSuite matching the Idris2 ABI tags (0-3).</summary>
public enum CipherSuite : byte
{
    AesGcm128Sha256 = 0,
    AesGcm256Sha384 = 1,
    ChaCha20Poly1305Sha256 = 2,
    AesCcm128Sha256 = 3
}

File: bindings/swift/Sources/ProvenServers/ProvenTls.swift
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// TLS protocol types for proven-servers.

/// TlsState matching the Idris2 ABI tags.
public enum TlsState: UInt8, CaseIterable, Sendable {
    case tlsIdle = 0
    case tlsClientHello = 1
    case tlsServerHello = 2
    case tlsNegotiating = 3
    case tlsEstablished = 4
    case tlsRenegotiating = 5
    case tlsShutdown = 6
}

/// TlsVersion matching the Idris2 ABI tags.
public enum TlsVersion: UInt8, CaseIterable, Sendable {
    case tls12 = 0
    case tls13 = 1
}

/// CipherSuite matching the Idris2 ABI tags.
public enum CipherSuite: UInt8, CaseIterable, Sendable {
    case aesGcm128Sha256 = 0
    case aesGcm256Sha384 = 1
    case chaCha20Poly1305Sha256 = 2
    case aesCcm128Sha256 = 3
}
